cpp/src/arrow/testing/gtest_util.h

```diff
@@ -94,7 +94,7 @@
 
 #define ARROW_EXPECT_OK(expr)                                           \
   do {                                                                  \
-    auto _res = (expr);                                                 \
+    const auto& _res = (expr);                                          \
```

Comment:
Are we sure it is ok to take a const-ref here? Presumably `(expr)` can return a rvalue.

Reply:
Everything still runs ok and I think the lifetime of the temporary is extended to the lifetime of the reference: https://stackoverflow.com/questions/42868744/what-happens-when-assigning-an-rvalue-to-const-auto

Full disclaimer, I did not recognize this myself, it was my IDE complaining.  I don't feel strongly about the change and am happy to revert back.

Reply:
Perhaps @bkietz can advise on this. On a more abstract standpoint, I'm not sure applying IDE-driven suggestions is necessarily a good strategy.

Reply:
On that abstract point, I also don't need to hold up the issue for unrelated changes.  I've reverted the change.